bundles/CoreBundle/Request/ParamConverter/DataObjectParamConverter.php

```diff
@@ -54,7 +55,7 @@ public function apply(Request $request, ParamConverter $configuration)
         $object = $class::getById($value);
         if (!$object) {
             throw new NotFoundHttpException(sprintf('Invalid data object ID given for parameter "%s".', $param));
-        } elseif (!$object->isPublished()) {
+        } elseif (!Tool::isFrontendRequestByAdmin() && !$object->isPublished()) {
```

Comment:
@dvesh3 We need to verify the login status in case `isFrontendRequestByAdmin() === true` otherwise you could expose confidential (not published) data just by adding one of those params: 
https://github.com/pimcore/pimcore/blob/2f18a0f26380c56b609d937b40d2050c8d1a3e1a/lib/Http/RequestHelper.php#L162

Reply:
fixed it by adding `Tool\Authentication::authenticateSession($request)` check 😉